src/ImageSharp/Formats/Jpeg/Components/Decoder/ColorConverters/JpegColorConverter.FromYCbCrVector.cs

```diff
@@ -5,36 +5,22 @@
 using System.Numerics;
 using System.Runtime.CompilerServices;
 using System.Runtime.InteropServices;
-
 using SixLabors.ImageSharp.Tuples;
 
 namespace SixLabors.ImageSharp.Formats.Jpeg.Components.Decoder.ColorConverters
 {
     internal abstract partial class JpegColorConverter
     {
-        internal sealed class FromYCbCrSimd : JpegColorConverter
+        internal sealed class FromYCbCrVector : VectorizedJpegColorConverter
         {
-            public FromYCbCrSimd(int precision)
-                : base(JpegColorSpace.YCbCr, precision)
+            public FromYCbCrVector(int precision)
+                : base(JpegColorSpace.YCbCr, precision, 8)
             {
             }
 
-            public override void ConvertToRgba(in ComponentValues values, Span<Vector4> result)
-            {
-                int remainder = result.Length % 8;
-                int simdCount = result.Length - remainder;
-                if (simdCount > 0)
-                {
-                    ConvertCore(values.Slice(0, simdCount), result.Slice(0, simdCount), this.MaximumValue, this.HalfValue);
-                }
+            protected override bool IsAvailable => true;
```

Comment:
```suggestion
            protected override bool IsAvailable => Vector.IsHardwareAccelerated && Vector<float>.Count == 4;
```

This is essentially an SSE path based on `Vector4`. The converter is not safe to run if the above conditions are not met, because of the `RoundAndDownscalePreVector8` call.

@tannergooding is there an environment variable to enforce a `(Vector.IsHardwareAccelerated && Vector<float>.Count == 4) == true` configuration for testing?

If not, I would suggest to remove this converter.

Reply:
If you set `COMPlus_EnableAVX2=0` it will force `Vector<T>` to be 16-bytes. We don't have a switch that explicitly controls the size of `Vector<T>` today.

Reply:
@tannergooding thanks!

@JimBobSquarePants https://github.com/SixLabors/ImageSharp/pull/1411#discussion_r518919719